pkg/kubelet/container/runtime_cache.go

```diff
@@ -80,6 +80,14 @@ func (r *runtimeCache) GetPods() ([]*Pod, error) {
 	return r.pods, nil
 }
 
+// getPods returns the current cached pods. This is used mainly for testing
+// because it does not start the refreshing thread.
+func (r *runtimeCache) getCachedPods() []*Pod {
+	r.Lock()
+	defer r.Unlock()
```

Comment:
It'd be a data race to read the pods without acquiring the lock if there is another thread updating the pods, which could happen if you call GetPods in your test.